diff --git a/CMakeLists.txt b/CMakeLists.txt
index 05a4af8533..91bb2fda98 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -693,6 +693,8 @@ include(gmxManageMuparser)
 
 include(gmxManageColvars)
 
+include(gmxManageLepton)
+
 include(gmxManagePlumed)
 
 include(gmxManageFMM)
diff --git a/src/gromacs/CMakeLists.txt b/src/gromacs/CMakeLists.txt
index 2025bc836e..33ab48924c 100644
--- a/src/gromacs/CMakeLists.txt
+++ b/src/gromacs/CMakeLists.txt
@@ -480,6 +480,8 @@ if (HAVE_MUPARSER)
 endif()
 gmx_manage_colvars()
 target_link_libraries(libgromacs PRIVATE colvars)
+gmx_manage_lepton()
+target_link_libraries(libgromacs PRIVATE lepton)
 target_link_libraries(libgromacs PRIVATE plumedgmx)
 
 # Make sure we fix "everything" found by compilers that support that
