diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8acf71e985..494254aac0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -691,6 +691,8 @@ include(gmxManageMuparser)
 
 include(gmxManageColvars)
 
+include(gmxManageLepton)
+
 include(gmxManagePlumed)
 
 ##################################################
diff --git a/src/gromacs/CMakeLists.txt b/src/gromacs/CMakeLists.txt
index d932509388..1e841b5b84 100644
--- a/src/gromacs/CMakeLists.txt
+++ b/src/gromacs/CMakeLists.txt
@@ -460,6 +460,8 @@ target_link_libraries(libgromacs PRIVATE lmfit)
 target_link_libraries(libgromacs PRIVATE muparser::muparser)
 gmx_manage_colvars()
 target_link_libraries(libgromacs PRIVATE colvars)
+gmx_manage_lepton()
+target_link_libraries(libgromacs PRIVATE lepton)
 target_link_libraries(libgromacs PRIVATE plumedgmx)
 
 # Make sure we fix "everything" found by compilers that support that
