diff --git a/cmake/gmxManageColvars.cmake b/cmake/gmxManageColvars.cmake
index 4f8b1eba98..735d9e752b 100644
--- a/cmake/gmxManageColvars.cmake
+++ b/cmake/gmxManageColvars.cmake
@@ -38,6 +38,19 @@ gmx_option_multichoice(GMX_USE_COLVARS
     INTERNAL NONE)
 mark_as_advanced(GMX_USE_COLVARS)
 
+function(gmx_set_colvars_torch)
+  find_package(Torch)
+  if (Torch_FOUND)
+    message(STATUS "Torch found, enabling for Colvars")
+    set_property(TARGET colvars_objlib PROPERTY CXX_STANDARD 17)
+    target_compile_definitions(colvars_objlib PRIVATE -DTORCH)
+    target_compile_options(colvars_objlib PRIVATE ${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS})
+    target_include_directories(colvars_objlib PRIVATE ${TORCH_INCLUDE_DIRS})
+    target_link_libraries(libgromacs PRIVATE "${TORCH_LIBRARIES}")
+  endif()
+endfunction()
+
+
 function(gmx_manage_colvars)
     if(GMX_USE_COLVARS STREQUAL "INTERNAL")
         # Create an object library for the colvars sources
@@ -68,5 +81,6 @@ function(gmx_manage_colvars)
 
         set(GMX_HAVE_COLVARS 0 CACHE INTERNAL "Is colvars found?")
     endif()
+    gmx_set_colvars_torch()
     mark_as_advanced(GMX_HAVE_COLVARS)
 endfunction()
