diff --git a/cmake/gmxManageColvars.cmake b/cmake/gmxManageColvars.cmake
index 9077822166..9a6ed2564d 100644
--- a/cmake/gmxManageColvars.cmake
+++ b/cmake/gmxManageColvars.cmake
@@ -54,6 +54,20 @@ function(gmx_manage_colvars)
         target_sources(colvars INTERFACE $<TARGET_OBJECTS:colvars_objlib>)
         target_include_directories(colvars SYSTEM INTERFACE $<BUILD_INTERFACE:${COLVARS_DIR}>)

+        if(GMX_OPENMP)
+            target_compile_options(colvars_objlib PRIVATE ${OpenMP_CXX_FLAGS})
+            target_link_libraries(colvars_objlib PRIVATE OpenMP::OpenMP_CXX)
+        endif()
+
+        find_package(Torch)
+        if (Torch_FOUND)
+          set_property(TARGET colvars PROPERTY CXX_STANDARD 17)
+          target_compile_definitions(colvars PRIVATE -DTORCH)
+          target_compile_options(colvars PRIVATE ${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS})
+          target_include_directories(colvars PRIVATE ${TORCH_INCLUDE_DIRS})
+          target_link_libraries(libgromacs PRIVATE "${TORCH_LIBRARIES}")
+        endif()
+
     else()
         # Create a dummy link target so the calling code doesn't need to know
         # whether colvars support is being compiled.
