diff --git a/src/gromacs/applied_forces/colvars/colvarproxygromacs.h b/src/gromacs/applied_forces/colvars/colvarproxygromacs.h
index 31ca7daf6e..32b1e8d0d7 100644
--- a/src/gromacs/applied_forces/colvars/colvarproxygromacs.h
+++ b/src/gromacs/applied_forces/colvars/colvarproxygromacs.h
@@ -97,8 +97,6 @@ protected:
 
 
 public:
-    friend class cvm::atom;
-
     /*! \brief Construct ColvarProxyGromacs from its parameters
      *
      * \param[in] colvarsConfigString Content of the colvars input file.
