diff --git a/src/gromacs/applied_forces/colvars/colvarsMDModule.cpp b/src/gromacs/applied_forces/colvars/colvarsMDModule.cpp
index 46df82f723..91e624c03c 100644
--- a/src/gromacs/applied_forces/colvars/colvarsMDModule.cpp
+++ b/src/gromacs/applied_forces/colvars/colvarsMDModule.cpp
@@ -56,6 +56,7 @@
 #include "gromacs/utility/mpicomm.h"
 #include "gromacs/utility/stringutil.h"

+#include "colvars_version.h"
 #include "colvarsforceprovider.h"
 #include "colvarsoptions.h"
 #include "colvarssimulationsparameters.h"
