diff --git a/config b/config
index ddf9f860..4148d87c 100755
--- a/config
+++ b/config
@@ -57,6 +57,8 @@ function error_syntax {
   echo '  --cuda-dlink arch=<arch>,code=<code> (for cuFFT, may be repeated)'
   echo '  --with-cuda-profiling (enables CUDA profiling with NVTX)'
   echo '  --with-rocm-profiling (enables ROCm profiling with ROCtracer API)'
+  echo '  --with-colvars-torch (enables torchann in Colvars)'
+  echo '  --torch-prefix <directory containing libTorch>'
   echo ''
   if [ -n "${PRINT_ARCH_LIST+set}" ]; then
     ARCH_PAT=''
@@ -152,6 +154,7 @@ function error_exists {
   use_mkl=0
   use_cuda=0
   use_hip=0
+  use_colvars_torch=0
   use_cuda_prof=0
   use_rocm_prof=0
   use_memopt=0
@@ -257,6 +260,19 @@ function error_exists {
         ARCH_SUFFIX_ARG=$ARCH_SUFFIX_ARG-$1
       ;;
 
+      --with-colvars-torch)
+        use_colvars_torch=1
+      ;;
+
+      --torch-prefix)
+        shift
+        if [ ! -d "$1" ]; then
+          echo "ERROR: No such directory $1"
+          error_syntax
+        fi
+        TORCH_PREFIX=$1
+      ;;
+
       --with-debug)
         use_debug=1
       ;;
@@ -1015,7 +1031,16 @@ function error_exists {
   if [ -n "$CC_OPTS" ]; then
     echo "COPTS = $CC_OPTS" >> Make.config
   fi
-
+  if (( "$use_colvars_torch" )); then
+    echo "TORCHDIR = $TORCH_PREFIX" >> Make.config
+    echo 'TORCHINCFLAGS = $(COPTI)$(TORCHDIR)/include $(COPTI)$(TORCHDIR)/include/torch/csrc/api/include' >> Make.config
+    echo 'EXTRACOLVARSFLAGS = -std=c++17 -DCOLVARS_TORCH $(TORCHINCFLAGS)' >> Make.config
+    if [[ $use_cuda && -f "${TORCH_PREFIX}/lib/libtorch_cuda.so" ]]; then 
+      echo 'EXTRALINKLIBS = -Wl,-rpath,$(TORCHDIR)/lib -L$(TORCHDIR)/lib -ltorch -ltorch_cpu -lc10 -ltorch_cuda -lc10_cuda' >> Make.config
+    else
+      echo 'EXTRALINKLIBS = -Wl,-rpath,$(TORCHDIR)/lib -L$(TORCHDIR)/lib -ltorch -ltorch_cpu -lc10' >> Make.config
+    fi
+  fi
 
   if (( $use_debug )); then
     echo 'CXXOPTS = -g' >> Make.config
