diff --git a/Makefile b/Makefile
index 1456d146..4bae1b7a 100644
--- a/Makefile
+++ b/Makefile
@@ -404,13 +404,7 @@ SBLIB = \
 
 include colvars/src/Makefile.namd
 
-LEPTONOBJS = \
-	$(DSTDIR)/CompiledExpression.o \
-	$(DSTDIR)/ExpressionProgram.o \
-	$(DSTDIR)/ExpressionTreeNode.o \
-	$(DSTDIR)/Operation.o \
-	$(DSTDIR)/ParsedExpression.o \
-	$(DSTDIR)/Parser.o 
+include lepton/Makefile.namd
 
 # definitions for Charm routines
 CHARMC = $(CHARM)/bin/charmc
