diff --git a/Make.depends b/Make.depends
index 1f193c5f..29552c0c 100644
--- a/Make.depends
+++ b/Make.depends
@@ -3658,16 +3658,6 @@ obj/ComputeMap.o: \
 	$(CXX) $(CXXFLAGS) $(COPTO)obj/ComputeMap.o $(COPTC) src/ComputeMap.C
 obj/ComputeMgr.o: \
 	obj/.exists \
-	colvars/src/colvarmodule.h \
-	colvars/src/colvarmodule.h \
-	colvars/src/colvarproxy.h \
-	colvars/src/colvarproxy_io.h \
-	colvars/src/colvarproxy_replicas.h \
-	colvars/src/colvarproxy_system.h \
-	colvars/src/colvarproxy_tcl.h \
-	colvars/src/colvarproxy_volmaps.h \
-	colvars/src/colvartypes.h \
-	colvars/src/colvarvalue.h \
 	inc/BroadcastMgr.decl.h \
 	inc/ComputeCUDAMgr.decl.h \
 	inc/ComputeMgr.decl.h \
@@ -3768,7 +3758,6 @@ obj/ComputeMgr.o: \
 	src/ComputeTclBC.h \
 	src/ComputeThole.h \
 	src/ComputeThole.inl \
-	src/ConfigList.h \
 	src/Controller.h \
 	src/CudaComputeGBISKernel.h \
 	src/CudaComputeNonbonded.h \
@@ -3847,7 +3836,6 @@ obj/ComputeMgr.o: \
 	src/Priorities.h \
 	src/ProcessorPrivate.h \
 	src/ProxyMgr.h \
-	src/Random.h \
 	src/ReadAmberParm.h \
 	src/ReductionMgr.h \
 	src/ResizeArray.h \
@@ -3870,7 +3858,6 @@ obj/ComputeMgr.o: \
 	src/UniqueSortedArray.h \
 	src/Vector.h \
 	src/WorkDistrib.h \
-	src/colvarproxy_namd.h \
 	src/common.h \
 	src/dlloader.h \
 	src/fstream_namd.h \
@@ -6414,6 +6401,33 @@ obj/GlobalMasterTMD.o: \
 	src/parm.h \
 	src/structures.h
 	$(CXX) $(CXXFLAGS) $(COPTO)obj/GlobalMasterTMD.o $(COPTC) src/GlobalMasterTMD.C
+obj/GlobalMasterColvars.o: \
+	obj/.exists \
+	colvars/src/colvarmodule.h \
+	colvars/src/colvarmodule.h \
+	colvars/src/colvarproxy.h \
+	colvars/src/colvarproxy_io.h \
+	colvars/src/colvarproxy_replicas.h \
+	colvars/src/colvarproxy_system.h \
+	colvars/src/colvarproxy_tcl.h \
+	colvars/src/colvarproxy_volmaps.h \
+	colvars/src/colvartypes.h \
+	colvars/src/colvarvalue.h \
+	src/CudaRecord.h \
+	src/GlobalMaster.h \
+	src/GlobalMasterColvars.C \
+	src/GlobalMasterColvars.h \
+	src/HipDefines.h \
+	src/Lattice.h \
+	src/NamdTypes.h \
+	src/ResizeArray.h \
+	src/ResizeArrayRaw.h \
+	src/Tensor.h \
+	src/Vector.h \
+	src/colvarproxy_namd.h \
+	src/colvarproxy_namd_version.h \
+	src/common.h
+	$(CXX) $(CXXFLAGS) $(COPTO)obj/GlobalMasterColvars.o $(COPTC) src/GlobalMasterColvars.C
 obj/Matrix4Symmetry.o: \
 	obj/.exists \
 	src/CudaRecord.h \
@@ -6630,147 +6644,55 @@ obj/colvarproxy_namd.o: \
 	colvars/src/colvarparams.h \
 	colvars/src/colvarparse.h \
 	colvars/src/colvarproxy.h \
-	colvars/src/colvarproxy.h \
 	colvars/src/colvarproxy_io.h \
 	colvars/src/colvarproxy_replicas.h \
 	colvars/src/colvarproxy_system.h \
 	colvars/src/colvarproxy_tcl.h \
 	colvars/src/colvarproxy_volmaps.h \
-	colvars/src/colvarscript.h \
-	colvars/src/colvarscript_commands.h \
-	colvars/src/colvarscript_commands_bias.h \
-	colvars/src/colvarscript_commands_colvar.h \
 	colvars/src/colvartypes.h \
 	colvars/src/colvarvalue.h \
 	colvars/src/colvarvalue.h \
-	inc/BroadcastMgr.decl.h \
-	inc/ComputeCUDAMgr.decl.h \
-	inc/ComputeMgr.decl.h \
 	inc/DataExchanger.decl.h \
-	inc/GlobalGPUMgr.decl.h \
-	inc/LdbCoordinator.decl.h \
-	inc/NamdCentLB.decl.h \
-	inc/NamdDummyLB.decl.h \
-	inc/NamdHybridLB.decl.h \
 	inc/Node.decl.h \
-	inc/PatchData.decl.h \
-	inc/PatchMgr.decl.h \
-	inc/ProxyMgr.decl.h \
-	inc/SynchronousCollectives.decl.h \
-	inc/SynchronousCollectives.def.h \
 	plugins/include/molfile_plugin.h \
 	plugins/include/vmdplugin.h \
-	src/AVXTiles.h \
-	src/Array.h \
-	src/AtomMap.h \
 	src/BOCgroup.h \
-	src/BackEnd.h \
-	src/Box.h \
-	src/BroadcastClient.h \
-	src/BroadcastMgr.h \
-	src/BroadcastObject.h \
-	src/CollectiveDeviceBuffer.h \
-	src/Compute.h \
-	src/ComputeAniso.h \
-	src/ComputeAniso.inl \
-	src/ComputeBondedCUDA.h \
-	src/ComputeBondedCUDAKernel.h \
-	src/ComputeCUDAMgr.h \
-	src/ComputeConsForceCUDA.h \
-	src/ComputeGridForceCUDA.h \
-	src/ComputeGroupRestraintsCUDA.h \
-	src/ComputeHomeTuples.h \
-	src/ComputeLonepairsCUDA.h \
-	src/ComputeMap.h \
-	src/ComputeMgr.h \
-	src/ComputeNonbondedUtil.h \
-	src/ComputeOneFourNbTholes.h \
-	src/ComputeRestraintsCUDA.h \
-	src/ComputeSMDCUDA.h \
-	src/ComputeSelfTuples.h \
-	src/ComputeThole.h \
-	src/ComputeThole.inl \
 	src/ConfigList.h \
 	src/Controller.h \
-	src/CudaComputeGBISKernel.h \
-	src/CudaComputeNonbonded.h \
-	src/CudaComputeNonbondedKernel.h \
-	src/CudaComputeNonbondedKernel.hip.h \
-	src/CudaNonbondedTables.h \
-	src/CudaPmeSolverUtil.h \
-	src/CudaPmeSolverUtilKernel.h \
 	src/CudaRecord.h \
-	src/CudaTileListKernel.h \
-	src/CudaTileListKernel.hip.h \
-	src/CudaUtils.h \
 	src/DataExchanger.h \
 	src/Debug.h \
-	src/GlobalGPUMgr.h \
 	src/GlobalMaster.h \
-	src/GlobalMasterServer.h \
+	src/GlobalMasterColvars.h \
 	src/GridForceGrid.h \
 	src/GridForceGrid.inl \
-	src/GridforceGridCUDA.h \
-	src/GridforceGridCUDAKernel.h \
 	src/GromacsTopFile.h \
 	src/GroupRestraintsParam.h \
 	src/HipDefines.h \
-	src/HomePatch.h \
-	src/HomePatchList.h \
 	src/Hydrogen.h \
 	src/InfoStream.h \
 	src/Lattice.h \
-	src/LdbCoordinator.h \
 	src/MGridforceParams.h \
-	src/MShakeKernel.h \
 	src/MStream.h \
-	src/MathArray.h \
-	src/MigrateAtomsMsg.h \
-	src/Migration.h \
-	src/MigrationBondedCUDAKernel.h \
-	src/MigrationCUDAKernel.h \
 	src/Molecule.h \
 	src/NamdState.h \
 	src/NamdTypes.h \
 	src/Node.h \
-	src/OwnerBox.h \
 	src/PDB.h \
 	src/PDBData.h \
-	src/Parameters.h \
-	src/Patch.h \
-	src/PatchData.h \
-	src/PatchMap.h \
-	src/PatchMap.inl \
-	src/PatchMgr.h \
-	src/PatchTypes.h \
-	src/PmeBase.h \
-	src/PmeSolverUtil.h \
 	src/Pointer.h \
-	src/Priorities.h \
 	src/ProcessorPrivate.h \
-	src/ProxyMgr.h \
 	src/Random.h \
 	src/ReadAmberParm.h \
 	src/ReductionMgr.h \
 	src/ResizeArray.h \
-	src/ResizeArrayIter.h \
 	src/ResizeArrayRaw.h \
 	src/ScriptTcl.h \
-	src/Sequencer.h \
-	src/SequencerCUDA.h \
-	src/SequencerCUDAKernel.h \
-	src/Settle.h \
 	src/SimParameters.h \
 	src/SortableResizeArray.h \
-	src/SortedArray.h \
-	src/SynchronousCollectives.h \
 	src/Tensor.h \
-	src/TupleString.h \
-	src/TupleTypesCUDA.h \
 	src/UniqueSet.h \
-	src/UniqueSetIter.h \
 	src/UniqueSetRaw.h \
-	src/UniqueSortedArray.h \
 	src/Vector.h \
 	src/colvarproxy_namd.C \
 	src/colvarproxy_namd.h \
@@ -6780,11 +6702,68 @@ obj/colvarproxy_namd.o: \
 	src/imd.h \
 	src/main.h \
 	src/parm.h \
-	src/strlib.h \
-	src/structures.h \
-	src/xxh3.h \
-	src/xxhash.h
+	src/structures.h
 	$(CXX) $(COLVARSCXXFLAGS) $(COPTO)obj/colvarproxy_namd.o $(COPTC) src/colvarproxy_namd.C
+obj/colvarproxy_cudaglobalmaster.o: \
+	obj/.exists \
+	colvars/src/colvar_rotation_derivative.h \
+	colvars/src/colvaratoms.h \
+	colvars/src/colvardeps.h \
+	colvars/src/colvarmodule.h \
+	colvars/src/colvarparams.h \
+	colvars/src/colvarparse.h \
+	colvars/src/colvarparse.h \
+	colvars/src/colvarproxy.h \
+	colvars/src/colvarproxy.h \
+	colvars/src/colvarproxy_io.h \
+	colvars/src/colvarproxy_replicas.h \
+	colvars/src/colvarproxy_system.h \
+	colvars/src/colvarproxy_tcl.h \
+	colvars/src/colvarproxy_volmaps.h \
+	colvars/src/colvarscript.h \
+	colvars/src/colvarscript_commands.h \
+	colvars/src/colvarscript_commands_bias.h \
+	colvars/src/colvarscript_commands_colvar.h \
+	colvars/src/colvartypes.h \
+	colvars/src/colvartypes.h \
+	colvars/src/colvarvalue.h \
+	plugins/include/molfile_plugin.h \
+	plugins/include/vmdplugin.h \
+	src/CudaGlobalMasterClient.h \
+	src/CudaRecord.h \
+	src/CudaUtils.h \
+	src/GridForceGrid.h \
+	src/GromacsTopFile.h \
+	src/GroupRestraintsParam.h \
+	src/HipDefines.h \
+	src/Hydrogen.h \
+	src/InfoStream.h \
+	src/Lattice.h \
+	src/MGridforceParams.h \
+	src/MStream.h \
+	src/Molecule.h \
+	src/NamdTypes.h \
+	src/PDB.h \
+	src/PDBData.h \
+	src/ReadAmberParm.h \
+	src/ResizeArray.h \
+	src/ResizeArrayRaw.h \
+	src/ScriptTcl.h \
+	src/SimParameters.h \
+	src/SortableResizeArray.h \
+	src/Tensor.h \
+	src/UniqueSet.h \
+	src/UniqueSetRaw.h \
+	src/Vector.h \
+	src/colvarproxy_cudaglobalmaster.C \
+	src/colvarproxy_cudaglobalmaster.h \
+	src/colvarproxy_cudaglobalmaster_kernel.h \
+	src/common.h \
+	src/fstream_namd.h \
+	src/imd.h \
+	src/parm.h \
+	src/structures.h
+	$(CXX) $(CXXFLAGS) $(COPTO)obj/colvarproxy_cudaglobalmaster.o $(COPTC) src/colvarproxy_cudaglobalmaster.C
 obj/GridForceGrid.o: \
 	obj/.exists \
 	inc/Node.decl.h \
@@ -10859,7 +10838,7 @@ obj/MigrationBondedCUDAKernel.o: \
 	src/TupleTypesCUDA.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/MigrationBondedCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`MigrationBondedCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/MigrationBondedCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`MigrationBondedCUDAKernel.cu
 obj/MigrationCUDAKernel.o: \
 	obj/.exists \
 	src/CudaComputeNonbondedKernel.h \
@@ -10879,7 +10858,7 @@ obj/MigrationCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/MigrationCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`MigrationCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/MigrationCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`MigrationCUDAKernel.cu
 obj/SequencerCUDAKernel.o: \
 	obj/.exists \
 	src/CudaRecord.h \
@@ -10898,7 +10877,7 @@ obj/SequencerCUDAKernel.o: \
 	src/Vector.h \
 	src/common.h \
 	src/structures.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/SequencerCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`SequencerCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/SequencerCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`SequencerCUDAKernel.cu
 obj/ComputeBondedCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeBondedCUDAKernel.cu \
@@ -10915,7 +10894,7 @@ obj/ComputeBondedCUDAKernel.o: \
 	src/TupleTypesCUDA.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeBondedCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeBondedCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeBondedCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeBondedCUDAKernel.cu
 obj/ComputeLonepairsCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeLonepairsCUDA.h \
@@ -10926,14 +10905,14 @@ obj/ComputeLonepairsCUDAKernel.o: \
 	src/HipDefines.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeLonepairsCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeLonepairsCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeLonepairsCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeLonepairsCUDAKernel.cu
 obj/ComputePmeCUDAKernel.o: \
 	obj/.exists \
 	src/ComputePmeCUDAKernel.cu \
 	src/ComputePmeCUDAKernel.h \
 	src/CudaUtils.h \
 	src/HipDefines.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputePmeCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputePmeCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputePmeCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputePmeCUDAKernel.cu
 obj/ComputeGridForceCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeGridForceCUDAKernel.cu \
@@ -10950,7 +10929,7 @@ obj/ComputeGridForceCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeGridForceCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGridForceCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeGridForceCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGridForceCUDAKernel.cu
 obj/ComputeRestraintsCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeRestraintsCUDAKernel.cu \
@@ -10965,7 +10944,7 @@ obj/ComputeRestraintsCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeRestraintsCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeRestraintsCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeRestraintsCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeRestraintsCUDAKernel.cu
 obj/ComputeConsForceCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeConsForceCUDAKernel.cu \
@@ -10980,7 +10959,7 @@ obj/ComputeConsForceCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeConsForceCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeConsForceCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeConsForceCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeConsForceCUDAKernel.cu
 obj/ComputeGlobalMasterVirialCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeGlobalMasterVirialCUDAKernel.cu \
@@ -10995,7 +10974,7 @@ obj/ComputeGlobalMasterVirialCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeGlobalMasterVirialCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGlobalMasterVirialCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeGlobalMasterVirialCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGlobalMasterVirialCUDAKernel.cu
 obj/ComputeSMDCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeCOMCudaKernel.h \
@@ -11011,7 +10990,7 @@ obj/ComputeSMDCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeSMDCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeSMDCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeSMDCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeSMDCUDAKernel.cu
 obj/ComputeGroupRes1GroupCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeCOMCudaKernel.h \
@@ -11027,7 +11006,7 @@ obj/ComputeGroupRes1GroupCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeGroupRes1GroupCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGroupRes1GroupCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeGroupRes1GroupCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGroupRes1GroupCUDAKernel.cu
 obj/ComputeGroupRes2GroupCUDAKernel.o: \
 	obj/.exists \
 	src/ComputeCOMCudaKernel.h \
@@ -11043,7 +11022,7 @@ obj/ComputeGroupRes2GroupCUDAKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/ComputeGroupRes2GroupCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGroupRes2GroupCUDAKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/ComputeGroupRes2GroupCUDAKernel.o $(COPTC) `$(NATIVEPATH) src/`ComputeGroupRes2GroupCUDAKernel.cu
 obj/CudaComputeGBISKernel.o: \
 	obj/.exists \
 	src/ComputeGBIS.inl \
@@ -11054,7 +11033,7 @@ obj/CudaComputeGBISKernel.o: \
 	src/CudaUtils.h \
 	src/DeviceCUDA.h \
 	src/HipDefines.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaComputeGBISKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaComputeGBISKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaComputeGBISKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaComputeGBISKernel.cu
 obj/CudaComputeNonbondedKernel.o: \
 	obj/.exists \
 	src/CudaComputeNonbondedInteractions.h \
@@ -11066,7 +11045,7 @@ obj/CudaComputeNonbondedKernel.o: \
 	src/CudaUtils.h \
 	src/DeviceCUDA.h \
 	src/HipDefines.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaComputeNonbondedKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaComputeNonbondedKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaComputeNonbondedKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaComputeNonbondedKernel.cu
 obj/CudaComputeNonbondedKernelHip.o: \
 	obj/.exists \
 	src/CudaComputeNonbondedInteractions.h \
@@ -11089,7 +11068,7 @@ obj/CudaComputeNonbondedKernelHip.o: \
 	src/Vector.h \
 	src/common.h \
 	src/imd.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaComputeNonbondedKernelHip.o $(COPTC) `$(NATIVEPATH) src/`CudaComputeNonbondedKernelHip.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaComputeNonbondedKernelHip.o $(COPTC) `$(NATIVEPATH) src/`CudaComputeNonbondedKernelHip.cu
 obj/CudaPmeSolverUtilKernel.o: \
 	obj/.exists \
 	src/CudaPmeSolverUtilKernel.cu \
@@ -11097,7 +11076,7 @@ obj/CudaPmeSolverUtilKernel.o: \
 	src/CudaRecord.h \
 	src/CudaUtils.h \
 	src/HipDefines.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaPmeSolverUtilKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaPmeSolverUtilKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaPmeSolverUtilKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaPmeSolverUtilKernel.cu
 obj/CudaTileListKernel.o: \
 	obj/.exists \
 	src/CudaComputeNonbondedKernel.h \
@@ -11108,7 +11087,7 @@ obj/CudaTileListKernel.o: \
 	src/CudaUtils.h \
 	src/DeviceCUDA.h \
 	src/HipDefines.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaTileListKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaTileListKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaTileListKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaTileListKernel.cu
 obj/CudaTileListKernelHip.o: \
 	obj/.exists \
 	src/CudaComputeNonbondedKernel.hip.h \
@@ -11119,13 +11098,13 @@ obj/CudaTileListKernelHip.o: \
 	src/CudaUtils.h \
 	src/DeviceCUDA.h \
 	src/HipDefines.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaTileListKernelHip.o $(COPTC) `$(NATIVEPATH) src/`CudaTileListKernelHip.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaTileListKernelHip.o $(COPTC) `$(NATIVEPATH) src/`CudaTileListKernelHip.cu
 obj/DeviceCUDAkernel.o: \
 	obj/.exists \
 	src/CudaUtils.h \
 	src/DeviceCUDAkernel.cu \
 	src/HipDefines.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/DeviceCUDAkernel.o $(COPTC) `$(NATIVEPATH) src/`DeviceCUDAkernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/DeviceCUDAkernel.o $(COPTC) `$(NATIVEPATH) src/`DeviceCUDAkernel.cu
 obj/MShakeKernel.o: \
 	obj/.exists \
 	src/CudaUtils.h \
@@ -11133,7 +11112,7 @@ obj/MShakeKernel.o: \
 	src/MShakeKernel.cu \
 	src/MShakeKernel.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/MShakeKernel.o $(COPTC) `$(NATIVEPATH) src/`MShakeKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/MShakeKernel.o $(COPTC) `$(NATIVEPATH) src/`MShakeKernel.cu
 obj/CudaGlobalMasterServerKernel.o: \
 	obj/.exists \
 	src/CudaGlobalMasterClient.h \
@@ -11150,7 +11129,7 @@ obj/CudaGlobalMasterServerKernel.o: \
 	src/Tensor.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaGlobalMasterServerKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaGlobalMasterServerKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaGlobalMasterServerKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaGlobalMasterServerKernel.cu
 obj/CudaGlobalMasterClientKernel.o: \
 	obj/.exists \
 	src/CudaGlobalMasterClientKernel.cu \
@@ -11159,7 +11138,13 @@ obj/CudaGlobalMasterClientKernel.o: \
 	src/HipDefines.h \
 	src/Vector.h \
 	src/common.h
-	$(CUDACC) $(CUDACCOPTS) $(COPTO)obj/CudaGlobalMasterClientKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaGlobalMasterClientKernel.cu
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/CudaGlobalMasterClientKernel.o $(COPTC) `$(NATIVEPATH) src/`CudaGlobalMasterClientKernel.cu
+obj/colvarproxy_cudaglobalmaster_kernel.o: \
+	obj/.exists \
+	src/HipDefines.h \
+	src/colvarproxy_cudaglobalmaster_kernel.cu \
+	src/colvarproxy_cudaglobalmaster_kernel.h
+	$(CUDACC) $(CUDACCOPTS) $(COPTI)$(COLVARSINCDIR) $(COPTO)obj/colvarproxy_cudaglobalmaster_kernel.o $(COPTC) `$(NATIVEPATH) src/`colvarproxy_cudaglobalmaster_kernel.cu
 obj/dcdplugin.o: \
 	obj/.exists \
 	plugins/include/molfile_plugin.h \
