diff --git a/Make.depends b/Make.depends
index 994b3914..1d54373f 100644
--- a/Make.depends
+++ b/Make.depends
@@ -6398,6 +6398,45 @@ obj/GlobalMasterTMD.o: \
 	src/fitrms.h \
 	src/Debug.h
 	$(CXX) $(CXXFLAGS) $(COPTO)obj/GlobalMasterTMD.o $(COPTC) src/GlobalMasterTMD.C
+obj/GlobalMasterColvars.o: \
+	obj/.exists \
+	src/GlobalMasterColvars.C \
+	src/Controller.h \
+	src/common.h \
+	src/Vector.h \
+	src/Tensor.h \
+	src/Lattice.h \
+	src/NamdTypes.h \
+	src/CudaRecord.h \
+	src/HipDefines.h \
+	src/ResizeArray.h \
+	src/ResizeArrayRaw.h \
+	src/fstream_namd.h \
+	src/DataExchanger.h \
+	src/main.h \
+	inc/DataExchanger.decl.h \
+	src/Pointer.h \
+	src/GlobalMaster.h \
+	src/GlobalMasterColvars.h \
+	src/colvarproxy_namd.h \
+	src/colvarproxy_namd_version.h \
+	src/SimParameters.h \
+	src/imd.h \
+	src/MGridforceParams.h \
+	src/GroupRestraintsParam.h \
+	src/Random.h \
+	src/ConfigList.h \
+	colvars/src/colvarmodule.h \
+	colvars/src/colvarproxy.h \
+	colvars/src/colvarmodule.h \
+	colvars/src/colvartypes.h \
+	colvars/src/colvarproxy_io.h \
+	colvars/src/colvarproxy_replicas.h \
+	colvars/src/colvarproxy_system.h \
+	colvars/src/colvarproxy_tcl.h \
+	colvars/src/colvarproxy_volmaps.h \
+	colvars/src/colvarvalue.h
+	$(CXX) $(CXXFLAGS) $(COPTO)obj/GlobalMasterColvars.o $(COPTC) src/GlobalMasterColvars.C
 obj/Matrix4Symmetry.o: \
 	obj/.exists \
 	src/Matrix4Symmetry.C \
@@ -6637,6 +6676,8 @@ obj/colvarproxy_namd.o: \
 	src/MStream.h \
 	plugins/include/molfile_plugin.h \
 	plugins/include/vmdplugin.h \
+	src/GlobalMasterColvars.h \
+	src/GlobalMaster.h \
 	src/GridForceGrid.inl \
 	src/PDB.h \
 	src/parm.h \
